jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3TreeTable.kt

```diff
@@ -57,14 +56,14 @@ class S3TreeTable(
 
             val directoryKey = node.getDirectoryKey()
 
-            virtualFiles.map {
-                s3Client.upload(project, it.inputStream, it.length, bucketVirtual.name, directoryKey + it.name).whenComplete { _, error ->
-                    when (error) {
-                        is Throwable -> error.notifyError(message("s3.upload.object.failed", it.name))
-                        else -> {
-                            invalidateLevel(node)
-                            refresh()
-                        }
+            GlobalScope.launch {
```

Comment:
"Application code usually should use an application-defined CoroutineScope. Using async or launch on the instance of GlobalScope is highly discouraged."

Should we create a scope for the tree tabel?

Reply:
OK - good feedback. I'm new to co-routines and was just following tutorials online :P